packages/common/src/repository/categoryRepository.ts

```diff
@@ -0,0 +1,44 @@
+import { SelectedFields } from 'drizzle-orm/pg-core/query-builders/select.types';
+import { SQL } from 'drizzle-orm/sql';
+import { isEmpty } from 'lodash-es';
+
+import { Dependencies } from '../config/config.js';
+import { Category, NewCategory, dictionaryCategories } from '../models/dictionary_categories.js';
+
+const repository = (dependencies: Dependencies) => {
+	return {
+		save: async (data: NewCategory): Promise<Category> => {
+			const { db, logger } = dependencies;
+			try {
+				const savedCategory = await db.insert(dictionaryCategories).values(data).returning();
+				logger.info(`Category ${data.name} saved successfully`);
+				return savedCategory[0];
+			} catch (error) {
+				logger.error(`Failed saving category ${data.name}. Details: ${error}`);
+				throw error;
+			}
+		},
+
+		select: async (
+			selectionFields: SelectedFields | undefined,
+			conditions: SQL<unknown> | ((aliases: SelectedFields) => SQL<unknown> | undefined) | undefined,
+		): Promise<
+			| Category[]
+			| {
+					[x: string]: unknown;
+			  }[]
```

Comment:
not a you think nor anything you have to fix, but this styling looks super weird lol

Reply:
the idea of using `{[x: string]: unknown;}` here is to be able to return whatever the user request on the select command. for example with Drizzle ORM we can request column alias:`db.select({
  field1: users.id,
  field2: users.name,
}).from(category);`  in that case we intend to receive an object with fields {field1: '', field2: ''}. not a category object itself